Remove PLOG rates

jsantner opened the pull request:
Added code to remove PLOG reaction rates from a chemkin mechanism and replace with single or double Arrhenius reaction rates using `scipy.optimize.curve_fit`. Comparisons between original and fitted rates can be plotted. This includes tests for individual rates, as well as a test comparing original and modified mechanism results for a constant pressure reactor simulation.

This code is based on the plog branch at [this point in time](https://github.com/kyleniemeyer/mech_util/tree/d4ca22ffc44eef4af428a82d80c18c91acb476fb), so it can't automatically merge. I'm not sure how to fix that - maybe you need to make a fork from that version, and then I can put this pull request there?

**Unresolved Issue**
`test_output.py` is failing. This code tests `convert_mech_irrev`, but I don't think I changed anything related to that function. Maybe I copied the wrong `mech.dat` and `mech_blessed.dat`? I don't want to make any changes in that portion of the code, it's really outside the scope of this pull request.

jsantner committed:
Add the files I've been working on in a separate, private repository

In the next few commits, I'll integrate these files

jsantner committed:
Move file to correct directory

jsantner committed:
Update mech_util.py

jsantner committed:
First attempt to deal with pressure units correctly, not tested yet.

jsantner committed:
Lots of small debugging changes, mostly unit-related

- Add __str__ to reaction class
- Comment out tests that I want to ignore for now
- Use included write_mech module
- Follow through with units

jsantner committed:
Minor fixes, mostly related to error/warning messages

- Improve printing of reactions
- fix directory mistake in test
- Improve error/warning messages
- improve documentation

jsantner committed:
Improve testing resolution

Also improve printing of time history from testing.

jsantner committed:
Minor formatting changes

jsantner committed:
Test that all PLOG rates are removed.

jsantner committed:
Add more tests

Mostly taken from original plog tests in mech_util

jsantner committed:
Improve code style and docstrings

jsantner committed:
Add a published test mechanism

jsantner committed:
Adding manual test

Strange results, seems to be independent of end time? Work on this

jsantner committed:
Adjust fictional reaction rates

jsantner committed:
Remove unused test. Turn irrev_mech tests back on

jsantner committed:
Final changes

While looking at the diff with the original branch, I made some minor changes to clean things up. Renamed a file to remove my initials, deleted an old file, and other minor changes.

jsantner committed:
Update manual_test.py

kyleniemeyer commented:
Great @jsantner, thanks! I will take a look at this soon and manually resolve the conflicts.

jsantner committed:
Allow user to control amount of plotting

User can decide to create figures for every PLOG fit, or only those with large error. Also, minor change to warning message.
